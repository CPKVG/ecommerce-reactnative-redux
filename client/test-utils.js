// import { configureStore } from '@reduxjs/toolkit';
// import React from 'react';
// import { render as rtlRender } from '@testing-library/react-native'
// import { Provider } from 'react-redux';
// import { PersistGate } from 'redux-persist/integration/react';
// import { persistor } from './redux/createStore';
// import { rootReducer } from './redux/rootReducer';

// //custom render

// function render(
//     ui ,{
//         preloadedState,
//         store = configureStore({ reducer: rootReducer, preloadedState}),
//         ...renderOptions
//     } = {}
// ){
//   function Wrapper( {childern} ){
//       return <Provider store = {store}>
//           <PersistGate persistor={persistor}>
//             {childern}
//           </PersistGate>
//       </Provider>
//   }  
//   return rtlRender(ui, {wrapper: Wrapper, ...renderOptions})
// }

// //re-export everything
// export * from '@testing-library/react-native'
// //override render method
// export { render }


import React from "react"
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
// import configureStore from 'redux-mock-store';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { rootReducer } from "./redux/rootReducer";

const INITIAL_STATE = { ...rootReducer }; // Or use value from root reducer
const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)
// const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
  render(ui, {
    
    wrapper: ({ children }) => <
      Provider store={store}>
          {children}
      </Provider>,
    ...options
  });

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { reduxRender as render, store };