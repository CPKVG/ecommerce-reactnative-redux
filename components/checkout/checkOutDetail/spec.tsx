import React, { useState } from "react"
import CheckOutDetail from "."
import renderer from "react-test-renderer"
import { render } from "../../../test-utils"
import { FlatList, SectionList, Text, TextInput, View } from "react-native"
import { cleanup, fireEvent } from "@testing-library/react-native"
// import { mount, shallow } from 'enzyme'




// import { render } from "@testing-library/react-native"
const DATA = [
    {
        title:"title1",
        data:["dataA1","dataB1","dataC1"]
    },
    {
        title:"title2",
        data:["dataA2","dataB2","dataC2"]
    },
    {
        title:"title3",
        data:["dataA3","dataB3","dataC3"]
    },
]

afterEach(cleanup)
jest.useFakeTimers()
describe("checkout detail component",()=>{

    test("it should not allow numbers inputed",() => {
        
    })

    test("it should return error/red hightlight if creds do not match",() => {
         
    })

    test("it should highlight input green for valid inputs",()=>{

    })

    test("it should dispatch and return value via selector",()=>{
        
    })

    //snapshot test
    test("render input correctly on start", async() => {
        const tree = render(<CheckOutDetail/>).toJSON
        expect(tree).toMatchSnapshot();
    })

    it("renders the sectionList component", () => {
        const tree = renderer.create(
        <View>
        <SectionList
            sections = {DATA}
            renderItem = {({item}:any) => {
                return(
                    <View>
                        <FlatList
                            data = {item}
                            numColumns={2}
                            renderItem = {({item}:any) => {
                                return(
                                <View>
                                    <TextInput
                                        placeholder = {item}
                                    />
                                </View>
                            )}}
                        />
                    </View>
            
                )
            }}
            renderSectionHeader={({ section: { title } }) => (
                <Text>{title}</Text>
            )}
            
            />
        </View>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })

})

