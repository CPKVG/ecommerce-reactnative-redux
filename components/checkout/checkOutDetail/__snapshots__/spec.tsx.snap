// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkout detail component render input correctly on start 1`] = `[Function]`;

exports[`checkout detail component renders the sectionList component 1`] = `
<View>
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [
            "dataA1",
            "dataB1",
            "dataC1",
          ],
          "title": "title1",
        },
        Object {
          "data": Array [
            "dataA2",
            "dataB2",
            "dataC2",
          ],
          "title": "title2",
        },
        Object {
          "data": Array [
            "dataA3",
            "dataB3",
            "dataC3",
          ],
          "title": "title3",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
        5,
      ]
    }
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text>
          title1
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            data="dataA1"
            getItem={[Function]}
            getItemCount={[Function]}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="d"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="a"
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="t"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="a"
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="A"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="1"
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            data="dataB1"
            getItem={[Function]}
            getItemCount={[Function]}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="d"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="a"
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="t"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="a"
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="B"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="1"
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            data="dataC1"
            getItem={[Function]}
            getItemCount={[Function]}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="d"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="a"
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="t"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="a"
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="C"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="1"
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text>
          title2
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            data="dataA2"
            getItem={[Function]}
            getItemCount={[Function]}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="d"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="a"
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="t"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="a"
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="A"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="2"
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            data="dataB2"
            getItem={[Function]}
            getItemCount={[Function]}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="d"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="a"
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="t"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="a"
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="B"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="2"
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            data="dataC2"
            getItem={[Function]}
            getItemCount={[Function]}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="d"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="a"
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="t"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="a"
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View>
                  <TextInput
                    placeholder="C"
                  />
                </View>
                <View>
                  <TextInput
                    placeholder="2"
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
